mesh/mesh_readers.cpp

```diff
@@ -426,6 +426,28 @@ void Mesh::CreateVTKMesh(const Vector &points, const Array<int> &cell_data,
       j = cell_offsets[i];
    }
 
+   // determine spaceDim based on min/max differences detected each dimension
+   spaceDim = 0;
+   if (np > 0)
+   {
+      double min_value, max_value;
+      for (int d = 3; d > 0; --d)
+      {
+         min_value = max_value = points(3*0 + d-1);
+         for (int i = 1; i < np; i++)
+         {
+            min_value = std::min(min_value, points(3*i + d-1));
+            max_value = std::max(max_value, points(3*i + d-1));
+            if (min_value != max_value)
```

Comment:
Is it appropriate to ignore the possibility of roundoff error here?

Reply:
Yes, I think so. If the mesh is 2D, then all z-coordinates are set to some fixed value (usually 0). I don't know under what circumstances the user would perform arithmetic operations on the z-coordinate leading to round-off errors when the mesh is 2D.

Reply:
I was thinking of the mesh generator itself. I don't like to trust black boxes...

But you're probably right.